src/FrameReflower/Text.php

```diff
@@ -73,81 +74,131 @@ public function __construct(TextFrameDecorator $frame, FontMetrics $fontMetrics)
-        $current_line = $this->_block_parent->get_current_line_box();
+        $current_line = $block->get_current_line_box();
+        $fontMetrics = $this->getFontMetrics();
 
         // Determine the available width
-        $line_width = $this->_frame->get_containing_block("w");
+        $line_width = $frame->get_containing_block("w");
         $current_line_width = $current_line->left + $current_line->w + $current_line->right;
-
         $available_width = $line_width - $current_line_width;
 
-        // Account for word-spacing
-        $word_spacing = (float)$style->length_in_pt($style->word_spacing);
-        $char_spacing = (float)$style->length_in_pt($style->letter_spacing);
+        // Account for word and letter spacing
+        $word_spacing = (float) $style->length_in_pt($style->word_spacing);
+        $char_spacing = (float) $style->length_in_pt($style->letter_spacing);
 
         // Determine the frame width including margin, padding & border
-        $visible_text = preg_replace('/\xAD/u', '', $text);
-        $text_width = $this->getFontMetrics()->getTextWidth($visible_text, $font, $size, $word_spacing, $char_spacing);
-        $mbp_width =
-            (float)$style->length_in_pt([$style->margin_left,
-                $style->border_left_width,
-                $style->padding_left,
-                $style->padding_right,
-                $style->border_right_width,
-                $style->margin_right], $line_width);
+        $visible_text = preg_replace('/\xAD/u', "", $text);
+        $text_width = $fontMetrics->getTextWidth($visible_text, $font, $size, $word_spacing, $char_spacing);
+        $mbp_width = (float) $style->length_in_pt([
+            $style->margin_left,
+            $style->border_left_width,
+            $style->padding_left,
+            $style->padding_right,
+            $style->border_right_width,
+            $style->margin_right
+        ], $line_width);
 
         $frame_width = $text_width + $mbp_width;
 
-// Debugging:
-//    Helpers::pre_r("Text: '" . htmlspecialchars($text). "'");
-//    Helpers::pre_r("width: " .$frame_width);
-//    Helpers::pre_r("textwidth + delta: $text_width + $mbp_width");
-//    Helpers::pre_r("font-size: $size");
-//    Helpers::pre_r("cb[w]: " .$line_width);
-//    Helpers::pre_r("available width: " . $available_width);
-//    Helpers::pre_r("current line width: " . $current_line_width);
+        // Debugging:
+        //     Helpers::pre_r("Text: '" . htmlspecialchars($text). "'");
+        //     Helpers::pre_r("width: " .$frame_width);
+        //     Helpers::pre_r("textwidth + delta: $text_width + $mbp_width");
+        //     Helpers::pre_r("font-size: $size");
+        //     Helpers::pre_r("cb[w]: " .$line_width);
+        //     Helpers::pre_r("available width: " . $available_width);
+        //     Helpers::pre_r("current line width: " . $current_line_width);
 
-//     Helpers::pre_r($words);
+        //     Helpers::pre_r($words);
```

Comment:
I feel like it's time to start removing "debugging" code that's commented out. If it's not conditionally enabled by the debug flags then we probably don't need it.

Reply:
Done.